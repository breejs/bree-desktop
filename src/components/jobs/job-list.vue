<script setup>
import JobListItem from './job-list-item.vue';

defineProps({
  jobs: {
    type: [Array, undefined],
    required: true,
    default: () => []
  },
  allowEdit: {
    type: Boolean,
    default: false
  }
});

defineEmits(['start', 'stop', 'restart']);
</script>

<template lang="pug">
ul.list-group.list-group-flush.pe-0
  template(v-if='jobs?.length > 0')
    JobListItem(
      v-for='job in jobs',
      :key='job.name',
      :job='job',
      :allowEdit='allowEdit',
      @start='$emit("start", $event)',
      @stop='$emit("stop", $event)',
      @restart='$emit("restart", $event)'
    )

  template(v-else)
    li.list-group-item
      = 'No jobs'
</template>
