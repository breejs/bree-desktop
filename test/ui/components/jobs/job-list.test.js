import { app } from '@storybook/vue3';
import { test, expect } from 'vitest';
import { storyTests } from '../../_story-tests';

import * as stories from '@/components/jobs/job-list.stories';

test('expect directive to exist', () => {
  expect(Object.keys(app._context.directives)).toEqual(['tooltip']);
});

storyTests(stories);
