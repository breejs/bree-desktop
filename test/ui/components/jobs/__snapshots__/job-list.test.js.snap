// Vitest Snapshot v1

exports[`story snapshots > NoJobs 1`] = `
"<ul class=\\"list-group list-group-flush\\">
  <li class=\\"list-group-item\\">No jobs</li>
</ul>"
`;
